scripts/publish-metadata.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+import sys
+import requests
+import json
+import subprocess
+
+DESCRIPTION = """
+Uploads a given block of metadata to the Terra Data Catalog
+
+To run this script you must authed as an "admin" on the
+ specified environment's catalog resource
+
+"""
+
+
+def generate_access_token(user):
+    auth_as_user(user)
+
+    print_proc = subprocess.Popen(
+        ["gcloud auth print-access-token"], stdout=subprocess.PIPE, shell=True
+    )
+    (out, err) = print_proc.communicate()
+
+    return out.decode("ASCII").strip()
+
+
+def auth_as_user(user):
+    proc = subprocess.Popen(
+        [f"gcloud auth login {user} --brief"], stdout=subprocess.PIPE, shell=True
+    )
+    proc.communicate()
+
+
+def create_dataset(env, access_token, storage_system, storage_source_id, entry):
+    headers = {
+        "Authorization": f"Bearer {access_token}",
+        "Content-Type": "application/json",
+    }
+
+    request = {
+        "storageSystem": str(storage_system),
+        "storageSourceId": str(storage_source_id),
+        "catalogEntry": json.dumps(entry),
+    }
+    response = requests.post(
```

Comment:
I highly recommend any `requests` installation occurs in a `virtualenv`, it avoids many of the issues with system packages. See the other scripts for how to do this.

Reply:
I can add a comment similar to what Christine did. I think that helpfully explains how to set up a venv while allowing users to use their own mechanisms to use `requests` if they would like to do otherwise.